example/ROS_demo/face_detector_viewer.cpp

```diff
@@ -0,0 +1,152 @@
+            camera_topic, 10, 
+            std::bind(&FaceDetectorViewer::imageCallback, this, std::placeholders::_1));
+
+        face_detected_sub_ = this->create_subscription<std_msgs::msg::Int32>(
+            face_detected_topic, 10, 
+            std::bind(&FaceDetectorViewer::faceDetectedCallback, this, std::placeholders::_1));
+
+        face_roi_sub_ = this->create_subscription<sensor_msgs::msg::RegionOfInterest>(
+            face_roi_topic, 10, 
+            std::bind(&FaceDetectorViewer::faceRoiCallback, this, std::placeholders::_1));
+
+        face_landmarks_sub_ = this->create_subscription<geometry_msgs::msg::PointStamped>(
+            face_landmarks_topic, 10, 
+            std::bind(&FaceDetectorViewer::faceLandmarksCallback, this, std::placeholders::_1));
+
+        // Create OpenCV window
+        cv::namedWindow("Face Detection Viewer", cv::WINDOW_NORMAL);
+        cv::resizeWindow("Face Detection Viewer", 640, 480);
+    }
+
+    ~FaceDetectorViewer()
+    {
+        cv::destroyAllWindows();
+    }
+
+private:
+    void imageCallback(const sensor_msgs::msg::Image::SharedPtr msg)
+    {
+        try
+        {
+            current_frame_ = cv_bridge::toCvShare(msg, "bgr8")->image.clone();
+            updateDisplay();
+        }
+        catch (cv_bridge::Exception& e)
+        {
+            RCLCPP_ERROR(this->get_logger(), "cv_bridge exception: %s", e.what());
+        }
+    }
+
+    void faceDetectedCallback(const std_msgs::msg::Int32::SharedPtr msg)
+    {
+        face_detected_ = msg->data;
+        updateDisplay();
+    }
+
+    void faceRoiCallback(const sensor_msgs::msg::RegionOfInterest::SharedPtr msg)
+    {
+        face_roi_ = cv::Rect(msg->x_offset, msg->y_offset, msg->width, msg->height);
+        updateDisplay();
+    }
+
+    void faceLandmarksCallback(const geometry_msgs::msg::PointStamped::SharedPtr msg)
+    {
+        cv::Point landmark(msg->point.x, msg->point.y);
+        face_landmarks_.push_back(landmark);
+        if (face_landmarks_.size() > 6) {
+            face_landmarks_.erase(face_landmarks_.begin());
+        }
+        updateDisplay();
+    }
+
+    void updateDisplay()
+    {
+        if (current_frame_.empty()) return;
+
+        cv::Mat display_frame = current_frame_.clone();
+
+        // Draw face detection status
+        std::string top_left_text = "Detected: " + std::to_string(face_detected_);
+        cv::putText(display_frame, top_left_text, cv::Point(20, 70), cv::FONT_HERSHEY_PLAIN, 3, cv::Scalar(0, 0, 255), 2);
+
+        // Draw face ROI
+        if (face_roi_.width > 0 && face_roi_.height > 0) {
+            cv::rectangle(display_frame, face_roi_, cv::Scalar(0, 255, 0), 2);
+        }
+
+        // Draw face landmarks
+        for (const auto& landmark : face_landmarks_) {
+            cv::circle(display_frame, landmark, 2, cv::Scalar(0, 255, 0), -1);
+        }
+
+        // Display the frame
+        // cv::setBackend(cv::CAP_GSTREAMER);        
```

Comment:
I would highly suggest not leaving commented code, unless it serves purpose

Reply:
ok